test/builtin/files_io/test_files.py

```diff
@@ -80,6 +80,262 @@ def test_close():
     ), f"temporary filename {temp_filename} should not appear"
 
 
+@pytest.mark.parametrize(
+    ("str_expr", "msgs", "str_expected", "fail_msg"),
+    [
+        ('Close["abc"]', ("abc is not open.",), "Close[abc]", None),
+        (
+            "exp = Sin[1]; FilePrint[exp]",
+            ("File specification Sin[1] is not a string of one or more characters.",),
+            "FilePrint[Sin[1]]",
+            None,
+        ),
+        (
+            'FilePrint["somenonexistentpath_h47sdmk^&h4"]',
+            ("Cannot open somenonexistentpath_h47sdmk^&h4.",),
+            "FilePrint[somenonexistentpath_h47sdmk^&h4]",
+            None,
+        ),
+        (
+            'FilePrint[""]',
+            ("File specification  is not a string of one or more characters.",),
+            "FilePrint[]",
+            None,
+        ),
+        (
+            'Get["SomeTypoPackage`"]',
+            ("Cannot open SomeTypoPackage`.",),
+            "$Failed",
+            None,
+        ),
+        ## Parser Tests
+        (
+            "Hold[<< ~/some_example/dir/] // FullForm",
+            None,
+            'Hold[Get["~/some_example/dir/"]]',
+            None,
+        ),
+        (
+            r"Hold[<<`/.\-_:$*~?] // FullForm",
+            None,
+            r'Hold[Get["`/.\\\\-_:$*~?"]]',
+            None,
+        ),
+        (
+            "OpenRead[]",
+            ("OpenRead called with 0 arguments; 1 argument is expected.",),
+            "OpenRead[]",
+            None,
+        ),
+        (
+            "OpenRead[y]",
+            ("File specification y is not a string of one or more characters.",),
+            "OpenRead[y]",
+            None,
+        ),
+        (
+            'OpenRead[""]',
+            ("File specification  is not a string of one or more characters.",),
+            "OpenRead[]",
+            None,
+        ),
+        (
+            'OpenRead["MathicsNonExampleFile"]',
```

Comment:
@mmatera There are a problems with this test.

It assumes this file does not exist. Another test in this parameterize loop creates a file with exactly this name.

Instead, the test should ensure that a file does not exist is used. One way to do this is to create a _new_ named temporary directory which is empty and then look for some name there. https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory can help here.

So probably this should be removed from this parameterize loop and be put in a separate test that has the additional code